src/main/java/com/massivecraft/factions/data/MemoryFPlayer.java

```diff
@@ -1382,6 +1383,10 @@ public boolean isStealth() {
 
     @Override
     public void setStealth(boolean stealth) {
+        FPlayerToggleStealthEvent event = new FPlayerToggleStealthEvent(this, FPlayerToggleStealthEvent.Type.METHOD);
```

Comment:
You need to make sure that this wasn't called before calling this method. In the case a player toggles their stealth via the command, this event will be run twice.